Dockerfile

```diff
@@ -18,18 +18,23 @@ ENV NODE_VERSION 4.3.0
 
 RUN     wget https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz && \
           tar -zxf node-v$NODE_VERSION-linux-x64.tar.gz -C /usr/local && \
-          ln -sf node-v$NODE_VERSION-linux-x64 /usr/local/node && \
+          ln -sf /usr/local/node-v$NODE_VERSION-linux-x64 /usr/local/node && \
           ln -sf /usr/local/node/bin/npm /usr/local/bin/ && \
           ln -sf /usr/local/node/bin/node /usr/local/bin/ && \
           rm node-v$NODE_VERSION-linux-x64.tar.gz
 
-RUN     mkdir /usr/local/node/etc && cp /usr/local/etc/npmrc /usr/local/node/etc/npmrc
+RUN     touch /usr/local/etc/npmrc && \
+          mkdir /usr/local/node/etc && \
+          cp /usr/local/etc/npmrc /usr/local/node/etc/npmrc
 
 ENV     NODE_PATH /calypso/server:/calypso/client
 
 # Install base npm packages to take advantage of the docker cache
 COPY    ./package.json /calypso/package.json
-RUN     npm install --production
+COPY    ./npm-shrinkwrap.json /calypso/npm-shrinkwrap.json
+
+# Sometimes "npm install" fails the first time when the cache is empty, so we retry once if it failed
+RUN     npm install --production || npm install --production
```

Comment:
I couldn't reproduce the cache error in master. Could you try rebasing and removing the second try here?